cocotb/scheduler.py

```diff
@@ -585,27 +597,20 @@ def add(self, coroutine):
         useful error messages in the event of common gotchas.
         """
         if isinstance(coroutine, cocotb.decorators.coroutine):
-            self.log.critical(
-                "Attempt to schedule a coroutine that hasn't started")
-            coroutine.log.error("This is the failing coroutine")
-            self.log.warning(
-                "Did you forget to add parentheses to the @test decorator?")
-            self._test_result = TestError(
-                "Attempt to schedule a coroutine that hasn't started")
-            self._terminate = True
-            return
+            raise TypeError(
+                "Attempt to schedule a coroutine that hasn't started: {}.\n"
+                "Did you forget to add parentheses to the @test decorator?"
```

Comment:
`@test` -> `@cocotb.test()` would be more helpful.

Reply:
My intent was the leave the error messages untouched here, but I can make these changes if you want.

Reply:
That would be very much appreciated.